compressor/base.py

```diff
@@ -32,21 +31,36 @@ class Compressor(object):
     depending implementations details.
     """
 
-    def __init__(self, content=None, output_prefix=None,
+    output_prefix = 'compressed'
+    output_mimetypes = {}
+
+    def __init__(self, resource_kind, content=None, output_prefix=None,
                  context=None, filters=None, *args, **kwargs):
+        if filters is None:
+            filters = settings.COMPRESS_FILTERS[resource_kind]
         self.content = content or ""  # rendered contents of {% compress %} tag
-        self.output_prefix = output_prefix or "compressed"
+        if output_prefix:
+            self.output_prefix = output_prefix
         self.output_dir = settings.COMPRESS_OUTPUT_DIR.strip('/')
         self.charset = settings.DEFAULT_CHARSET
         self.split_content = []
         self.context = context or {}
-        self.type = output_prefix or ""
-        self.filters = filters or []
+        self.resource_kind = resource_kind
+        self.filters = filters
```

Comment:
`self.filters = filters or settings.COMPRESS_FILTERS[resource_kind]`?

Reply:
That would remove the option of removing filters altogether. I will put an `if None else ` in between.